Makefile

```diff
@@ -27,7 +27,6 @@ PLATFORMS ?= linux_amd64 linux_arm64
 # even though this repo doesn't build images (note the no-op img.build target below),
 # some of the init is needed for the cross build container, e.g. setting BUILD_REGISTRY
 -include build/makelib/image.mk
-img.build:
```

Comment:
When I gave the following a try:
- Clone the repo
- Remove the `img.build` target from the `Makefile`
- Run `make`
I get the following error:
```bash
❯ make
Submodule 'build' (https://github.com/upbound/build) registered for path 'build'
Cloning into '/private/tmp/upjet/build'...
Submodule path 'build': checked out 'bd5297bd16c113cbc5ed1905b1d96aa1cb3078ec'
Initial setup complete. Running make again . . .
11:02:25 [ .. ] verify go modules dependencies have expected content
all modules verified
11:02:27 [ OK ] go modules dependencies verified
11:02:28 [ .. ] go build linux_arm64
11:02:28 [ OK ] go build linux_arm64
make[4]: *** No rule to make target `img.build', needed by `build.artifacts.platform'.  Stop.
make[3]: *** [do.build.artifacts.linux_arm64] Error 2
make[2]: *** [build.all] Error 2
make[1]: *** [build] Error 2
make: *** [fallthrough] Error 2
```

My understanding is that this empty target was added to prevent this error.

Reply:
Looks like removing the include together with the empty target makes the default target behave as we would like:
```bash
diff --git a/Makefile b/Makefile
index cda2468..d1e3ffa 100644
--- a/Makefile
+++ b/Makefile
@@ -21,14 +21,6 @@ PLATFORMS ?= linux_amd64 linux_arm64
 # to run a target until the include commands succeeded.
 -include build/makelib/common.mk

-# ====================================================================================
-# Setup Images
-
-# even though this repo doesn't build images (note the no-op img.build target below),
-# some of the init is needed for the cross build container, e.g. setting BUILD_REGISTRY
--include build/makelib/image.mk
-img.build:
-
 # ====================================================================================
 # Setup Go
```

I will also leave a comment in the parent issue.

Reply:
Just left a comment discussing the root cause [here](https://github.com/crossplane/upjet/issues/361#issuecomment-1978477003). Moving the empty `img.build` target after to the `fallthrough` target also addresses the issue:
```bash
diff --git a/Makefile b/Makefile
index cda2468..f3732fa 100644
--- a/Makefile
+++ b/Makefile
@@ -21,14 +21,6 @@ PLATFORMS ?= linux_amd64 linux_arm64
 # to run a target until the include commands succeeded.
 -include build/makelib/common.mk
 
-# ====================================================================================
-# Setup Images
-
-# even though this repo doesn't build images (note the no-op img.build target below),
-# some of the init is needed for the cross build container, e.g. setting BUILD_REGISTRY
--include build/makelib/image.mk
-img.build:
-
 # ====================================================================================
 # Setup Go
 
@@ -60,6 +52,14 @@ fallthrough: submodules
        @echo Initial setup complete. Running make again . . .
        @make
 
+# ====================================================================================
+# Setup Images
+
+# even though this repo doesn't build images (note the no-op img.build target below),
+# some of the init is needed for the cross build container, e.g. setting BUILD_REGISTRY
+-include build/makelib/image.mk
+img.build:
+
 # Update the submodules, such as the common build scripts.
 submodules:
        @git submodule sync
```
but because we don't need the `build/makelib/image.mk` targets, we can safely remove the include as suggested above.

Reply:
Thanks @ulucinar - interesting that it works when the change is applied after already running `make` in the repo but it doesn't work on a clean repo.  I'll remove the `include`.